src/platform/plugins/shared/controls/public/controls/data_controls/range_slider/get_range_slider_control_factory.tsx

```diff
@@ -192,20 +192,25 @@ export const getRangesliderControlFactory = (): DataControlFactory<
         dataControlManager.api.dataViews$,
         dataControlManager.api.fieldName$,
         selections.value$,
+        min$,
+        max$,
       ])
         .pipe(debounceTime(0))
-        .subscribe(([dataViews, fieldName, value]) => {
+        .subscribe(([dataViews, fieldName, value, min, max]) => {
           const dataView = dataViews?.[0];
           const dataViewField =
             dataView && fieldName ? dataView.getFieldByName(fieldName) : undefined;
           const gte = parseFloat(value?.[0] ?? '');
           const lte = parseFloat(value?.[1] ?? '');
 
+          const hasValidLowerBound = !isNaN(gte) && gte !== min;
+          const hasValidUpperBound = !isNaN(lte) && lte !== max;
+          
           let rangeFilter: Filter | undefined;
-          if (value && dataView && dataViewField && !isNaN(gte) && !isNaN(lte)) {
+          if (value && dataView && dataViewField && (hasValidLowerBound || hasValidUpperBound)) {
             const params = {
-              gte,
-              lte,
+              gte: hasValidLowerBound ? gte : -Infinity,
+              lte: hasValidUpperBound ? lte : Infinity,
```

Comment:
I changed the logic here for allowing one sided range queries, e.g. { gte: 100} or { lte: 50}. It also ignores selections that match the min and max, but as a side effect, entering the min or max values in the range filter will not recognize the selection as values that can be applied when automatic changes are turned off. 

We avoid building filters unnecessarily this way, but is this the expected behavior here?
<img width="1039" height="652" alt="Screenshot 2025-08-08 at 4 21 48 PM" src="https://github.com/user-attachments/assets/72ee9984-c034-4e82-bf20-93684eaf7b4e" />

Reply:
I'm a bit confused what this is doing - I feel like the behaviour is the same as before? For example, if I enter `0` (the min value) in the bytes control, I get unsaved changes both with this change and without it:

<img width="1342" height="298" alt="image" src="https://github.com/user-attachments/assets/7888ac38-3f5a-4a0a-9e82-eb142830b94f" />

Can you give an example of what this is meant to change?